Assets/MXR.SDK/Runtime/Android/Utils/SafeJNI.cs

```diff
@@ -0,0 +1,186 @@
+    /// </summary>
+    public static class SafeJNI {
+        const string TAG = "SafeJNI";
+
+        /// <summary>
+        /// Calls a void-returning static method on a native object using method name and optional arguments. 
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <param name="methodName"></param>
+        /// <param name="args"></param>
+        /// <returns>Success: true. Failure: false</returns>
+        public static bool SafeCallStatic(this AndroidJavaObject obj, string methodName, params object[] args) {
+            if (obj == null) {
+                Debug.unityLogger.Log(LogType.Error, TAG, "JNI Error: Tried to call " + methodName + " on a null AndroidJavaObject");
+                return false;
+            }
+
+            try {
+                if (args == null || args.Length == 0)
+                    obj.CallStatic(methodName);
+                else
+                    obj.CallStatic(methodName, args);
+                return true;
+            }
+            catch (AndroidJavaException e) {
+                Debug.unityLogger.Log(LogType.Exception, TAG, "JNI Exception: " + e);
+                return false;
+            }
+            catch (Exception ex) {
+                Debug.unityLogger.Log(LogType.Exception, TAG, "Unexpected Exception: " + ex);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Calls a value-returning static method on a native object using method name and optional arguments. 
+        /// </summary>
+        /// <typeparam name="ReturnType">The type as which the result should be returned.</typeparam>
+        /// <param name="obj"></param>
+        /// <param name="methodName"></param>
+        /// <param name="args"></param>
+        /// <returns>Success: The field value. Failure: Default type value.</returns>
+        public static ReturnType SafeCallStatic<ReturnType>(this AndroidJavaObject obj, string methodName, params object[] args) {
+            if (obj == null) {
+                Debug.unityLogger.Log(LogType.Error, TAG, "JNI Error: Tried to call " + methodName + " on a null AndroidJavaObject");
+                return default;
+            }
+
+            try {
+                if (args == null || args.Length == 0)
+                    return obj.CallStatic<ReturnType>(methodName);
+                else
+                    return obj.CallStatic<ReturnType>(methodName, args);
+            }
+            catch (AndroidJavaException e) {
+                Debug.unityLogger.Log(LogType.Exception, TAG, "JNI Exception: " + e);
+                return default;
+            }
+            catch (Exception ex) {
+                Debug.unityLogger.Log(LogType.Exception, TAG, "Unexpected Exception: " + ex);
+                return default;
+            }
+        }
+
+        /// <summary>
+        /// Calls a void-returning non-static method on a native object using method name and optional arguments.
```

Comment:
Is this comment out dated ? void-returning

Reply:
No, it invokes `obj.CallStatic(methodName);` internally where methodName refers to a method that returns `void`.  Hence void-returning method.

On the other hand `obj.CallStatic<T>(methodName);` returns a value, hence it's a `value-returning` method.

But yes, the language needs more clarity here

Reply:
Made the codedoc comments clearer.